src/raftstore/store/worker/split_check.rs

```diff
@@ -24,31 +24,45 @@ use raftstore::coprocessor::CoprocessorHost;
 use raftstore::store::engine::{IterOption, Iterable};
 use raftstore::store::{keys, Callback, Msg};
 use raftstore::Result;
-use storage::{CfName, LARGE_CFS};
+use storage::{CfName, CF_WRITE, LARGE_CFS};
 use util::escape;
 use util::transport::{RetryableSendCh, Sender};
 use util::worker::Runnable;
 
 use super::metrics::*;
 
 #[derive(PartialEq, Eq)]
-struct KeyEntry {
+pub struct KeyEntry {
     key: Option<Vec<u8>>,
     pos: usize,
     value_size: usize,
+    cf: CfName,
 }
 
 impl KeyEntry {
-    fn new(key: Vec<u8>, pos: usize, value_size: usize) -> KeyEntry {
+    fn new(key: Vec<u8>, pos: usize, value_size: usize, cf: CfName) -> KeyEntry {
         KeyEntry {
             key: Some(key),
             pos,
             value_size,
+            cf,
         }
     }
 
     fn take(&mut self) -> KeyEntry {
-        KeyEntry::new(self.key.take().unwrap(), self.pos, self.value_size)
+        KeyEntry::new(self.key.take().unwrap(), self.pos, self.value_size, self.cf)
+    }
+
+    pub fn key(&self) -> &[u8] {
```

Comment:
Seems there is quite a bit of potential for danger here.

```rust
let entry = KeyEntry::new(..)
// ...
entry.take();
// ...
entry.key(); // Panics!
```

Perhaps `key()` can return an `Option`, or `take()` could consume `self`?